File: task_tips/app/src/main/java/com/example/task_tips/MainActivity.java
package com.example.task_tips;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Button;
import android.widget.Toast;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText orderAmountInput;
    private Spinner tipPercentageSpinner;
    private TextView totalAmountValue;
    private Button submitButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        // Найдем элементы
        orderAmountInput = findViewById(R.id.orderAmountInput);
        tipPercentageSpinner = findViewById(R.id.tipPercentageSpinner);
        totalAmountValue = findViewById(R.id.totalAmountValue);
        submitButton = findViewById(R.id.submitButton);

        // Обработчик для кнопки
        submitButton.setOnClickListener(v -> calculateTotal());

        // Обработчик для изменения значения в Spinner
        tipPercentageSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                calculateTotal(); // Автоматически пересчитываем сумму
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {}
        });

        // Код для изменения отступов
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    // Метод расчета итоговой суммы
    private void calculateTotal() {
        String orderAmountStr = orderAmountInput.getText().toString();

        if (orderAmountStr.isEmpty()) {
            totalAmountValue.setText("0.00 ₽");
            return;
        }

        double orderAmount = Double.parseDouble(orderAmountStr);

        // Получаем процент чаевых
        String selectedTip = tipPercentageSpinner.getSelectedItem().toString();
        int tipPercent = Integer.parseInt(selectedTip.replace("%", "")); // Убираем % и конвертируем в число

        // Рассчитываем итоговую сумму
        double total = orderAmount + (orderAmount * tipPercent / 100.0);

        // Устанавливаем результат
        totalAmountValue.setText(String.format("%.2f ₽", total));
    }
}

File: TrafficLights/app/src/main/java/com/example/trafficlights/MainActivity.kt
package com.example.trafficlights

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import android.widget.Button
import android.view.View


class TrafficLightActivity : AppCompatActivity() {

    private lateinit var red: View
    private lateinit var yellow: View
    private lateinit var green: View
    private var state = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        red = findViewById(R.id.redLight)
        yellow = findViewById(R.id.yellowLight)
        green = findViewById(R.id.greenLight)

        findViewById<Button>(R.id.btnChange).setOnClickListener {
            state = (state + 1) % 3
            updateLights()
        }

        updateLights()
    }

    private fun updateLights() {
        red.setBackgroundResource(if (state == 0) R.drawable.red_circle else R.drawable.gray_circle)
        yellow.setBackgroundResource(if (state == 1) R.drawable.yellow_circle else R.drawable.gray_circle)
        green.setBackgroundResource(if (state == 2) R.drawable.green_circle else R.drawable.gray_circle)
    }
}
